packages/zipkin/src/batch-recorder.js

```diff
@@ -1,28 +1,81 @@
 const {now, hrtime} = require('./time');
 const {Span, Endpoint} = require('./model');
 
-function PartialSpan(traceId) {
-  this.traceId = traceId;
-  this.startTimestamp = now();
-  this.startTick = hrtime();
-  this.delegate = new Span(traceId);
-  this.localEndpoint = new Endpoint({});
-}
-PartialSpan.prototype.finish = function finish() {
-  if (this.endTimestamp) {
-    return;
+/**
+ * default timeout = 60 seconds (in microseconds)
+ * @type {number}
+ */
+const defaultTimeout = 60 * 1000000;
+
+/**
+ * defaultTags property name
+ * @type {symbol}
+ */
+const defaultTagsSymbol = Symbol('defaultTags');
+
+/**
+ * @class PartialSpan
+ */
+class PartialSpan {
+  /**
+   * @constructor
+   * @param {TraceId} traceId
+   */
+  constructor(traceId) {
+    this.traceId = traceId;
+    this.startTimestamp = now();
+    this.startTick = hrtime();
+    this.delegate = new Span(traceId);
+    this.localEndpoint = new Endpoint({});
+  }
+
+  /**
+   * adds endTimestamp to span
+   * @return {void}
+   */
+  finish() {
+    if (this.endTimestamp) {
+      return;
+    }
+    this.endTimestamp = now(this.startTimestamp, this.startTick);
+  }
+
+  /**
+   * factory: creates new span and set
+   * @static
+   * @param {TraceId} id
+   * @param {Object} defaultTags
+   * @return {PartialSpan}
+   */
+  static create(id, defaultTags = {}) {
+    const span = new PartialSpan(id);
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const tag in defaultTags) {
+      if (defaultTags.hasOwnProperty(tag)) {
+        span.delegate.putTag(tag, defaultTags[tag]);
+      }
+    }
+
+    return span;
   }
-  this.endTimestamp = now(this.startTimestamp, this.startTick);
-};
+}
 
+/**
+ * @class BatchRecorder
+ */
 class BatchRecorder {
-  constructor({
-    logger,
-    timeout = 60 * 1000000 // default timeout = 60 seconds
-  }) {
+  /**
+   * @constructor
+   * @param {Object} options
+   * @property {HttpLogger|KafkaLogger} logger logs the data to openZipkin
```

Comment:
Minor nit, could we avoid this implementation detail `HttpLogger|KafkaLogger` unless it is absolutely necessary?

Reply:
well, I would love to have a real common interface for `HttpLogger|KafkaLogger` but we don't right now. I will fix with a `@typedef`.

Reply:
@jcchavezs if I replace by this, would it work?

```js
/**
 * @typedef {Object} ZipkinLogger
 * @property {function(Span):void} logSpan
 */

```

then use as:

```js
/**
 * @property {ZipkinLogger} logger logs the data to openZipkin
 */
```

Reply:
scratch that... I think we have a `ts` interface already...